parser/parser.go

```diff
@@ -537,21 +537,21 @@ func (p *parser) parseObjectRemainder(tok *token) (ast.Node, *token, error) {
 }
 
 /* parses for x in expr for y in expr if expr for z in expr ... */
-func (p *parser) parseComprehensionSpecs(end tokenKind) (*ast.ForSpec, *token, error) {
+func (p *parser) parseComprehensionSpecs(end tokenKind) (*ast.ForSpec, *ast.ForSpec, *token, error) {
```

Comment:
The added return is only used inside this function from recursive calls, right? Does it make sense to expose to an external caller?

Reply:
You're right. I thought it was not worth it to, since it's an internal thing used in two places. 

I've changed it, let me know if you think it's better.